CollegeRecords/src/main/java/dao/ExamDao.java

```diff
@@ -95,6 +97,30 @@ public Exam mapRow(ResultSet rs, int rowNum) throws SQLException {
         return result;
     }
 
+    public ExamStatistics queryExamStatistics(int examId) {
+        NamedParameterJdbcTemplate template = new NamedParameterJdbcTemplate(dataSource);
+
+        String sql = getQueryExamStatistics();
+        Map<String, String> params = new HashMap<>();
+        params.put("examId", valueOf(examId));
+        params.put("grade", "4");
+
+        ExamStatistics result = template.queryForObject(sql, params, new RowMapper<ExamStatistics>() {
+            @Override
+            public ExamStatistics mapRow(ResultSet rs, int rowNum) throws SQLException {
+                ExamStatistics statistics = new ExamStatistics();
+                statistics.setExam(queryExam(examId));
```

Comment:
Not recommended. This nesting of the database accesses makes the code harder to read/understand. One can expect here just a "mapping" of the result set fields to object properties, and miss the second query for the "exam" property.